File: test_data/code/python/1370-increasing-decreasing-string.py
from collections import Counter

class Solution:
    def sortString(self, s: str) -> str:
        cnt = Counter(s)
        res = []
        while cnt:
            # sort small characters
            temp = sorted([c for c in cnt if cnt[c] > 0])
            while temp and cnt[temp[0]] > 0:
                cnt[temp[0]] -= 1
                res.append(temp.pop(0))
            
            # sort large characters
            temp = sorted([c for c in cnt if cnt[c] > 0], reverse=True)
            while temp and cnt[temp[0]] > 0:
                cnt[temp[0]] -= 1
                res.append(temp.pop(0))

        return ''.join(res)

File: test_data/code/java/1370-increasing-decreasing-string.java
class Solution {
    public String sortString(String s) {
        char[] arr = s.toCharArray();
        Arrays.sort(arr);
        
        StringBuilder sb = new StringBuilder();
        int i = 0, j = arr.length - 1;
        
        while (i <= j) {
            if (arr[i] <= arr[j]) {
                sb.append(arr[i++]);
                if (i <= j) {
                    sb.append(arr[i--]);
                }
            } else {
                sb.append(arr[j--]);
                if (i <= j) {
                    sb.append(arr[i++]);
                }
            }
        }
        
        return sb.toString();
    }
}